test/extended/README.md

```diff
@@ -0,0 +1,145 @@
+LDAP, it can be `./test/extended/authentication`.  In order to have your test
+case executed by `./hack/test-extended/ldap/run.sh` you have to add `ldap:`
+prefix to the `Describe()` function:
+
+Example:
+```go
+var _ = g.Describe("ldap: Authenticate using LDAP", func() {
+  # ...
+})
+```
+
+
+CLI interface
+---------------------
+
+In order to be able to call the OpenShift CLI and Kubernetes and OpenShift REST client and simulate the OpenShift `oc` command in the test suit, first we need to create an instance of the CLI, in the top-level Ginkgo describe container. The top-level describe container shall also specify the the bucket, into which the test belongs and short test description. Other globally accessible variables can be declared(eg. fixtures) can be declared as well.
+
+```go
+package extended
+
+import (
+    g "github.com/onsi/ginkgo"
+    o "github.com/onsi/gomega"
+)
+
+var _ = g.Describe("<test bucket>: <Testing scenario>", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLI("test-name", exutil.KubeConfigPath())
+		testFixture = filepath.Join("fixtures", "test.json")
+	)
+})
+```
+
+The test suit shall be organized in lower-level Ginkgo describe container, together with an message which informs about the goal of the test. Inside the lower-level describe container specify a single spec with the `It` container , which shares the context in which the spec runs. The `It` container also takes a message, which informs how should be the goal achieved.
+
+```go
+var _ = g.Describe("default: STI build", func() {
+	defer GinkgoRecover()
+	var (
+		stiBuildFixture = filepath.Join("fixtures", "test-build.json")
+		oc              = exutil.NewCLI("build-sti", kubeConfigPath())
+	)
+
+	g.Describe("Building from a template", func() {
+		It(fmt.Sprintf("should create a image from %q template", stiBuildFixture), func() {
+			...
+		}
+	}
+}
+```
+
+After that you are free to simulate any `oc` command by calling the CLI methods from the extended package.
+
+As first, the command verb (get, create, start-build, ...) has to be specified upon the created CLI instance with the `Run()` method.
+```go
+oc = oc.Run("create")
+```
+
+Then the command parameters have to be specified by using the `Args()` command. You may also notice the methods can be easily chained.
+```go
+oc = oc.Run("create").Args("-f", testFixture)
+```
+
+A Go template can be set as a parameter for the OpenShift CLI command, by using the `Template()` method. Keep in mind that in order to use this method, the `get` verb has to be specified by the `Run()` command.
+```go
+oc = oc.Run("get").Template({{ .spec }})
+```
+is an equivalent to
+```go
+oc get foo -o template -t '{{ .spec }}
+```
+
+To execute the command you will need to call either `Execute()`, which will execute the command and return only error if any occurs, or `Output()` command which besides error also returns the output of the command in a form of a string.
```

Comment:
s/ only error if any occurs / any error that occurs /
s/ command which besides error also returns the output / which returns any error that occurs as well as the output /